### ISSUE TITLE
GitLab CI Failing even with the command running as expected

### ISSUE THREAD

**Issue by FabianoLothor** [maglnet/ComposerRequireChecker#365]
![image](https://user-images.githubusercontent.com/2192995/206824050-027f8f80-3e21-4a53-822f-8cb912008e25.png)

My CI script is failing without any error message, do you guys have any idea why?

```sh
docker compose exec -T my-app-name bash -c 'php -d xdebug.mode=off ./vendor/maglnet/composer-require-checker/bin/composer-require-checker check --output=json > composer-require-check-report.json'

// I already tried variations, always with the same result.

docker compose exec -T my-app-name php -d xdebug.mode=off ./vendor/maglnet/composer-require-checker/bin/composer-require-checker check --output=json > composer-require-check-report.json

docker compose exec -T my-app-name XDEBUG_MODE=off php ./vendor/maglnet/composer-require-checker/bin/composer-require-checker check --output=json > composer-require-check-report.json

docker compose exec -T my-app-name bash -c 'php -d xdebug.mode=off ./vendor/maglnet/composer-require-checker/bin/composer-require-checker check --output=json' > composer-require-check-report.json

docker compose exec -T my-app-name bash -c 'XDEBUG_MODE=off php ./vendor/maglnet/composer-require-checker/bin/composer-require-checker check --output=json > composer-require-check-report.json'
```

As mentioned, the command is executing correctly and the file is generated at the end.

---

The command works well locally (and apparently works well in the runner too, I just don't know why the job is failing.

**Comment by Ocramius** [comment]
Is the exit code perhaps overridden by an entrypoint in your container? 

Also, nothing in the artifacts either?

**Comment by FabianoLothor** [comment]
> Is the exit code perhaps overridden by an entrypoint in your container?

I'm not sure if I understood, but the container is using a `php` image with `tty` enabled, if it's the issue, I don't know.

The artifacts depend...

If I specify to always save I get the artifact perfectly.

```yml
    paths:
      - composer-require-check-report.json
    when: always
```

Specifying to get `on_success` I don't get any artifacts.

```yml
    paths:
      - composer-require-check-report.json
    when: on_success
```

**Comment by Ocramius** [comment]
Did you download them? What is in them?

**Comment by FabianoLothor** [comment]
It's just a json generated by the library.

**Comment by Ocramius** [comment]
So it contains a failure report 😛

**Comment by FabianoLothor** [comment]
Why? It isn't a report, it's as an artifact.

**Comment by FabianoLothor** [comment]
I'm using artifact.paths instead of artifact.reports.

**Comment by FabianoLothor** [comment]
btw, is it possible to save the generated file of this library as a JUnit report or something similar?

**Comment by Ocramius** [comment]
Please just attach the contents of the report: it probably just says that you have some violations, which is precisely why the exit code is non-zero (failure).

**Comment by FabianoLothor** [comment]
Oh, it makes sense.

I was expecting to get a file with requirement issues and have them available to check one by one

I probably will need to add an `allow_failure`.

**Comment by Ocramius** [comment]
Nack: use `|| true` if you want to build such semantics yourself.

Closing here meanwhile: clarified.
